08 - Fun with HTML5 Canvas/index-CLASS.html

```diff
@@ -0,0 +1,128 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>HTML5 Canvas</title>
+</head>
+  <body>
+    <canvas id="draw" width="800" height="800"></canvas>
+    <button class="clearCanvas">Clear Canvas</button>
+    <script>
+        //// classes
+        class DrawingCanvas {
+            constructor(id, contextID = '2d') {
+                this.canvas = document.querySelector(id);
+                this.context = this.canvas.getContext(contextID)
+                this.X = 0;
+                this.Y = 0;
+                this.hue = 0;
+                this.isDrawing = false;
+                this.isStrokeGrowing = true;
+
+                //// binding this to events
+                this.startDrawing = this.startDrawing.bind(this);
```

Comment:
yeah, this is annoying. personally, I prefer to use an arrow function in the when adding the event listener and then call the object's (class instance's) method directly. this way, you do not need this binding.

Reply:
Indeed! I guess it's good for me to try the basics first. Next time I try with arrow functions as well!